lib/uri.ml

```diff
@@ -1025,10 +981,12 @@ module Parser = struct
             ]))
 
   let host =
-    lift
-      (fun s -> Pct.decode (Pct.cast_encoded s))
-      (choice
-         [ reg_name; ipv4_address; ipv6_address (* | ipv4_literal TODO *) ])
+    choice
+      [ ipv4_address >>| (fun h -> `Ipv4_literal h)
+      ; ipv6_address >>| (fun h -> `Ipv6_literal h)
+      ; reg_name >>| (fun s -> `Host (Pct.decode (Pct.cast_encoded s)))
+      (* ipv4_literal TODO *)
```

Comment:
Not sure about this remaining TODO. From the RFC, we talk about `IP-literal` which includes `IPv6address` or `IPvFuture`. We probably should change it to: `TODO: IPvFuture` which seems to not be handled by this PR.